spopt/locate/base.py

```diff
@@ -1,3 +1,5 @@
+# ruff: noqa: B009, B010
```

Comment:
The ruff rules are assigned how? Just to know the reasons.

Reply:
This is a good question. The `ruff` config options can be found [here](https://docs.astral.sh/ruff/configuration/) and the rules can be found [here](https://docs.astral.sh/ruff/rules/). I am not sure there is a definitive guide, but my experience with these linters makes me want to be as granular as possible. This means:
* limiting super-global rules found in `pyproject.toml` to a bare minimum
* preferring in-line `noqa` to the file-level `noqa` rules if there are only a few (and this is ambiguous) instances
* using file-level `noqa` rules especially on legacy code.

I think for writing new code we should try to limit ourselves to in-line `noqa` where possible unless there is a good reason not to.

That's my 2-cents on the topic. Happy to hear yours, of course, @gegen07 & @martinfleis